proto/wtools/abase/l0/l1.test/Set.test.s

```diff
@@ -42,6 +42,241 @@ function dichotomy( test )
+    var src = new Set( [ 1, 1, 2, 2 ] );
+    var got = _.set[ env.method ]( src );
+    var expected = src;
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, a WeakSet`;
+    var obj1 = { a : 1, b : 2 };
+    var obj2 = { a : 3, b : 4 }
+    var src = new WeakSet( [ obj1, obj2 ] );
+    var got = _.set[ env.method ]( src );
+    var expected = src;
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, a WeakSet with user defined iterable at initialization`;
+    var obj1 = { a : 1, b : 2 };
+    var obj2 = { a : 3, b : 4 }
+    var src = new WeakSet( function* () { yield obj1, yield obj2 }() );
+    var got = _.set[ env.method ]( src );
+    var expected = src;
+    test.identical( got, expected );
+
+    /* */
+    test.case = `${__.entity.exportStringSolo( env )}, argument object`;
+    var src = arguments;
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an Array Prototype`;
+    var src = Array.prototype;
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an Array`;
+    var src = [ 1, 2, 3 ];
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, two array arguments`;
+    var src = [ 1, 2, 3 ];
+    var got = _.set[ env.method ]( src, src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, a typed Array`;
+    var src = new Uint8Array( 32 );
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, a hash map`;
+    var src = new Map( [ [ 1, 'one' ], [ 2, 'two' ], [ 3, 'three' ] ] );
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, a generator function`;
+    var src = function* ( i ) 
+    {
+      yield i;
+      yield i + 10;
+    }
+    var got = _.set[ env.method ]( src( 10 ) );
+    var expected = new Set( [ ... src( 10 ) ] );
+    test.identical( got, expected );
+
+    test.case = `${__.entity.exportStringSolo( env )}, an array having generator function as it's Symbol.iterator`;
+    var src = [];
+    src[ Symbol.iterator ] = function* ()
+    {
+      yield 1;
+      yield 2;
+      yield 3;
+    };
+    var got = _.set[ env.method ]( src );
+    var expected = new Set( [ ... src ]);
+    test.identical( got, expected );
+
+    //Throws error. [object GeneratorFunction] is not covered as a Symbol.iterator
```

Comment:
- Throws error
- [object GeneratorFunction] is not covered as a Symbol.iterator